lib/hutch/error_handlers/base.rb

```diff
@@ -0,0 +1,15 @@
+module Hutch
+  module ErrorHandlers
+    class Base
+      include Logging
+
+      def handle(properties, payload, consumer, ex)
+        raise NotImplementedError.new
```

Comment:
Unfortunately, `NotImplementedError` is subclassed from `ScriptError` which is subclassed from `Exception`.

I think it would be safer to use something that can be caught by `StandardError`.

Reply:
There seems to be no standard way of doing this besides the above, and not throwing an Error, but relying on the no method error being thrown when a subclass doesn't implement the method.

Do you suggest to define a custom Error class that inherits from StandardError?

Please cf. #297 for links to discussions regarding the above error... I'm a newbie to ruby (heh, that rhymed), so I really don't feel confident to make a call, due to the different opinions I found...

Also, we should probably fix [the ack base class](https://github.com/gocardless/hutch/blob/fa9383a40cfc4e681c4128e3cc47e84c8820546f/lib/hutch/acknowledgements/base.rb#L12) as well if we mind the above.

Reply:
`NotImplementedError` is right on the money in terms of what we'd like to communicate, though.

Reply:
Again, I don't really know, not a seasoned ruby programmer, but: the docs for the `NotImplementedError` can be interpreted in such a way that it is only to be used for instanced where a feature is not implemented on the current platform, due to a hardware / software deficit of the underlying system, not a missing implementation in the ruby code.
Even the error message in `error.c`, where the error class is defined, sounds as follow:
`[$FUNCTION_NAME() function is unimplemented on this machine](https://github.com/ruby/ruby/blob/1ae69a20645d7be556fd0ecd001b0bb5f3959330/error.c#L2338)`

Of course I don't actually get that message when I raise the error in irb, just get the text `NotImplementedError`...

The above are just meant as hints - you guys should decide how to handle this problem : )

Reply:
I don't think this discussion has to delay this PR, it's great :)

I have only called this out because in April this year we had apps not working due to a missed implementation, and we did not get any Honeybadger reports since it as `NotImplementedError` and zipped past through any layers of error reporting so we didn't know what was happening before looking at the logs.

Reply:
@sldblog the fact that in Ruby a `rescue` without specific exceptions doesn't catch everything not a problem library authors should worry about. Ruby designers made a mistake,  using a `rescue` without an exception class is therefore is not a good idea. `NotImplementedError` is exactly the exception that should be used in this case.